From 80def62c819d71e03e28bc504f7f1645f1de829d Mon Sep 17 00:00:00 2001
From: Armin Kuster <akuster808@gmail.com>
Date: Wed, 21 Jun 2023 07:46:38 -0400
Subject: [PATCH 2/5] standard.profile: expand checks

Upstream-Status: Pending
Signed-off-by: Armin Kuster <akuster808@gmail.com>

Upstream-status: Pending
---
 .../openembedded/profiles/standard.profile    | 52 +++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/products/openembedded/profiles/standard.profile b/products/openembedded/profiles/standard.profile
index 2d3a785647..72aac10c02 100644
--- a/products/openembedded/profiles/standard.profile
+++ b/products/openembedded/profiles/standard.profile
@@ -72,10 +72,13 @@ selections:
     - var_password_pam_remember_control_flag=required
     - var_password_pam_remember=5
     - set_password_hashing_algorithm_systemauth
+    - accounts_maximum_age_login_defs
     - var_accounts_maximum_age_login_defs=365
     - accounts_password_set_max_life_existing
+    - accounts_minimum_age_login_defs
     - var_accounts_minimum_age_login_defs=7
     - accounts_password_set_min_life_existing
+    - accounts_password_warn_age_login_defs
     - var_accounts_password_warn_age_login_defs=7
     - account_disable_post_pw_expiration
     - var_account_disable_post_pw_expiration=30
@@ -84,6 +87,7 @@ selections:
     - var_accounts_tmout=15_min
     - accounts_root_gid_zero
     - accounts_umask_etc_bashrc
+    - accounts_umask_etc_login_defs
     - use_pam_wheel_for_su
     - sshd_allow_only_protocol2
     - journald_forward_to_syslog
@@ -125,6 +129,8 @@ selections:
     - require_singleuser_auth
     - require_emergency_target_auth
     - disable_users_coredumps
+    - coredump_disable_backtraces
+    - coredump_disable_storage
     - configure_crypto_policy
     - var_system_crypto_policy=default_policy
     - dir_perms_world_writable_sticky_bits
@@ -147,6 +153,7 @@ selections:
     - file_permissions_backup_etc_group
     - file_groupowner_backup_etc_gshadow
     - file_owner_backup_etc_gshadow
+    - file_permissions_backup_etc_gshadow
     - file_permissions_unauthorized_world_writable
     - file_permissions_ungroupowned
     - accounts_root_path_dirs_no_write
@@ -159,7 +166,52 @@ selections:
     - account_unique_id
     - group_unique_id
     - group_unique_name
+    - kernel_module_sctp_disabled
+    - kernel_module_dccp_disabled
     - wireless_disable_interfaces
+    - sysctl_net_ipv4_ip_forward
+    - sysctl_net_ipv6_conf_all_forwarding
+    - sysctl_net_ipv6_conf_all_forwarding_value=disabled
+    - sysctl_net_ipv4_conf_all_send_redirects
+    - sysctl_net_ipv4_conf_default_send_redirects
+    - sysctl_net_ipv4_conf_all_accept_source_route
+    - sysctl_net_ipv4_conf_all_accept_source_route_value=disabled
+    - sysctl_net_ipv4_conf_default_accept_source_route
+    - sysctl_net_ipv4_conf_default_accept_source_route_value=disabled
+    - sysctl_net_ipv6_conf_all_accept_source_route
+    - sysctl_net_ipv6_conf_all_accept_source_route_value=disabled
+    - sysctl_net_ipv6_conf_default_accept_source_route
+    - sysctl_net_ipv6_conf_default_accept_source_route_value=disabled
+    - sysctl_net_ipv4_conf_all_accept_redirects
+    - sysctl_net_ipv4_conf_all_accept_redirects_value=disabled
+    - sysctl_net_ipv4_conf_default_accept_redirects
+    - sysctl_net_ipv4_conf_default_accept_redirects_value=disabled
+    - sysctl_net_ipv6_conf_all_accept_redirects
+    - sysctl_net_ipv6_conf_all_accept_redirects_value=disabled
+    - sysctl_net_ipv6_conf_default_accept_redirects
+    - sysctl_net_ipv6_conf_default_accept_redirects_value=disabled
+    - sysctl_net_ipv4_conf_all_secure_redirects
+    - sysctl_net_ipv4_conf_all_secure_redirects_value=disabled
+    - sysctl_net_ipv4_conf_default_secure_redirects
+    - sysctl_net_ipv4_conf_default_secure_redirects_value=disabled
+    - sysctl_net_ipv4_conf_all_log_martians
+    - sysctl_net_ipv4_conf_all_log_martians_value=enabled
+    - sysctl_net_ipv4_conf_default_log_martians
+    - sysctl_net_ipv4_conf_default_log_martians_value=enabled
+    - sysctl_net_ipv4_icmp_echo_ignore_broadcasts
+    - sysctl_net_ipv4_icmp_echo_ignore_broadcasts_value=enabled
+    - sysctl_net_ipv4_icmp_ignore_bogus_error_responses
+    - sysctl_net_ipv4_icmp_ignore_bogus_error_responses_value=enabled
+    - sysctl_net_ipv4_conf_all_rp_filter
+    - sysctl_net_ipv4_conf_all_rp_filter_value=enabled
+    - sysctl_net_ipv4_conf_default_rp_filter
+    - sysctl_net_ipv4_conf_default_rp_filter_value=enabled
+    - sysctl_net_ipv4_tcp_syncookies
+    - sysctl_net_ipv4_tcp_syncookies_value=enabled
+    - sysctl_net_ipv6_conf_all_accept_ra
+    - sysctl_net_ipv6_conf_all_accept_ra_value=disabled
+    - sysctl_net_ipv6_conf_default_accept_ra
+    - sysctl_net_ipv6_conf_default_accept_ra_value=disabled
     - package_firewalld_installed
     - service_firewalld_enabled
     - package_iptables_installed
-- 
2.43.0

